### PR TITLE
[Feature][Auth][Refactor 1] remove the heavy api call for listing tenants on check avail 

### PR DESCRIPTION
**Pull request by andxu**
…henticated

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java [hunk 1]
```diff
@@ -132,7 +128,7 @@ public void selectSubscription(List<String> selectedSubscriptionIds) {
     protected abstract Mono<TokenCredentialManager> createTokenCredentialManager();
 
     public Mono<Boolean> checkAvailable() {
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java:130]: Shall we add java doc here to indicate this method will throw auth exception if not available?

**Review by Flanker32 — APPROVED** [review]

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/core/devicecode/DeviceCodeAccount.java [hunk 2]
```diff
@@ -71,7 +71,7 @@ public String getClientId() {
     }
 
     public Mono<Account> continueLogin() {
-        return loginMono;
+        return loginMono.flatMap(ac -> super.continueLogin());
     }
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/core/devicecode/DeviceCodeAccount.java:76]: remove this method?

File: azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/AbstractMojoBase.java [hunk 3]
```diff
@@ -221,15 +221,15 @@ private static Account accountLogin(@Nonnull com.microsoft.azure.toolkit.lib.aut
...
+                return handleDeviceCodeAccount(com.microsoft.azure.toolkit.lib.Azure.az(AzureAccount.class).loginAsync(account, false).block());
             } else {
                 // user specify SP related configurations
                 return doServicePrincipalLogin(auth);
             }
         } else {
             // user specifies the auth type explicitly
             promptForOAuthOrDeviceCodeLogin(auth.getType());
-            return handleDeviceCodeAccount(com.microsoft.azure.toolkit.lib.Azure.az(AzureAccount.class).login(auth).account());
+            return handleDeviceCodeAccount(com.microsoft.azure.toolkit.lib.Azure.az(AzureAccount.class).loginAsync(auth, false).block());
```
Comment (wangmingliang-ms) [inline azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/AbstractMojoBase.java:232]: no need to use qualified name.

File: azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/AbstractMojoBase.java [hunk 4]
```diff
@@ -221,15 +221,15 @@ private static Account accountLogin(@Nonnull com.microsoft.azure.toolkit.lib.aut
                 }
                 // prompt if oauth or device code
                 promptForOAuthOrDeviceCodeLogin(account.getAuthType());
-                return handleDeviceCodeAccount(com.microsoft.azure.toolkit.lib.Azure.az(AzureAccount.class).login(account).account());
+                return handleDeviceCodeAccount(com.microsoft.azure.toolkit.lib.Azure.az(AzureAccount.class).loginAsync(account, false).block());
```
Comment (wangmingliang-ms) [inline azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/AbstractMojoBase.java:224]: why not use `login` directly?

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java [hunk 5]
```diff
@@ -132,7 +126,7 @@ public void selectSubscription(List<String> selectedSubscriptionIds) {
...
-        return loginStep1();
+        return preLoginCheck().doOnSuccess(avail -> this.entity.setAvailable(avail));
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java:129]: is there any difference between `checkAvailable` and the so-called `preLoginCheck`?

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java [hunk 6]
```diff
@@ -169,10 +168,12 @@ private void finishLogin() {
      * @return Mono = true if this account is available
      */
     private Mono<Boolean> loginStep1() {
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java:170]: method name should better describe what this method does, but not where this method is called.

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java [hunk 7]
```diff
@@ -154,12 +148,17 @@ public void selectSubscription(List<String> selectedSubscriptionIds) {
         });
     }
 
+    public Mono<Account> continueLogin() {
+        Azure.az(AzureAccount.class).setAccount(this);
+        return Mono.just(this);
+    }
+
     private void finishLogin() {
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java:156]: method name should better indicates what this method does, but not where this method is called.

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java [hunk 8]
```diff
@@ -395,15 +395,15 @@ private static Account accountLogin(@Nonnull com.microsoft.azure.toolkit.lib.aut
                 }
                 // prompt if oauth or device code
                 promptForOAuthOrDeviceCodeLogin(account.getAuthType());
-                return handleDeviceCodeAccount(com.microsoft.azure.toolkit.lib.Azure.az(AzureAccount.class).login(account).account());
+                return handleDeviceCodeAccount(com.microsoft.azure.toolkit.lib.Azure.az(AzureAccount.class).loginAsync(account, false).block());
```
Comment (wangmingliang-ms) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java:398]: 1. no need to use qualified name
2. consider make `Azure.az(AzureAccount.class)` a field.
3. why not use `login(account, false)`?

**Review by Flanker32 — APPROVED** [review]
Please rename loginStep1... to meaningful name

**Pull request merged**